Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -282,6 +288,12 @@ checkLibrary pkg lib =
     moduleDuplicates = dups (explicitLibModules lib ++
                              map moduleReexportName (reexportedModules lib))
 
+explicitIncludes :: L.HasBuildInfo a => a -> [FilePath]
+explicitIncludes x = view L.includes x ++ view L.installIncludes x
+
+explicitIncludes' :: L.HasBuildInfo a => a -> [FilePath]
```

Comment:
It's not immediately obvious what's the difference between `explicitIncludes'` and `explicitIncludes`. How about:
* `explicitIncludes' -> importedExplicitIncludes`
* `explicitIncludes  -> allExplicitIncludes`

?